website/blog/2022-12-13-using-cpts-and-taxonomies.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Using Custom Post Types and Taxonomies
+description: Explains how to register and use custom post types and taxonomies
+slug: using-cpts-and-taxonomies
+authors: obradovic
+date: 2022-12-13
+tags: [eightshift, boilerplate, cpt, custom post type, taxonomy, taxonomies, terms]
+hide_table_of_contents: false
+---
+
+By default, WordPress offers two post types for our content management - Posts and Pages. But sometimes, that is not enough. That's where custom post types and custom taxonomies come in. This blog post will cover the basics of registering CPTs and Taxonomies with Eightshift Dev kit.
+
+<!--truncate-->
+## Why should you use CPTs?
+The great thing about WordPress is how much extendability it offers. With projects becoming more complex, the need for additional post types and taxonomies increases. Sure, you can cram everything into default posts or pages, but this can become very chaotic. Using custom post types allows you to manage your content much better. You can separate galleries and projects from news articles etc.
+
+Taxonomies, like the default category, allow for even more grouping options.
+
+## Registering Custom Post Types
+For this exercise, we'll create a new CPT called `Projects`. To make the process of registering new CPTs as easy as possible, we'll use a WP-CLI command to create our Custom Post Type. To create your new CPT, enter the following command:
+
+```bash
+wp boilerplate create post-type --label='Project' --plural_label='Projects' --slug='project' --rewrite_url='project' --rest_endpoint_slug='projects'
+```
+
+> If your new CPT is not working, try flushing rewrite rules by re-saving the settings in **Settings -> Permalinks** or by using `wp cache flush` CLI command
+
+Your new post type is registered and ready to use! Easy, right?
+
+Sometimes all these parameters can be a bit confusing, so here's a quick reference of best practices when setting these parameters:
+
+| Parameter          | Singular/Plural | Writing style | Example  |
+|--------------------|-----------------|---------------|----------|
+| label              | Singular        | Regular       | Project  |
+| plural_label       | Plural          | Regular       | Projects |
+| slug               | Singular        | kebab-case    | project  |
+| rewrite_url        | Singular        | kebab-case    | project  |
+| rest_endpoint_slug | Plural          | kebab-case    | projects |
+
```

Comment:
We could add some explanation why this is the convention.

For instance, the slug will be attached to a _single_ custom post type in the database, which is why it should be a singular lower kebab-cased string.

The rest endpoint slug will expose a _collection_ of custom post types, which is why we want that to be a plural kebab-cased string.